doc/design/externally-manage-pf.md

```diff
@@ -0,0 +1,174 @@
+---
+title: Externally Manage PF
+authors:
+  - SchSeba
+reviewers:
+  - zeeke
+  - adrianchiris
+creation-date: 12-07-2023
+last-updated: 12-07-2023
+---
+
+# Externally Manage PF
+
+## Summary
+
+Allow the SR-IOV operator to configure and allocate a subset of virtual functions from a physical function not
+configured by the operator
+
+## Motivation
+
+The feature is needed to allow the operator to only configure a subset of virtual functions.
+This allows a third party component like nmstate, kubernetes-nmstate, NetworkManager to handle the creation
+and the usage of the virtual functions on the system.
+
+Before this change the SR-IOV operator is the only component that should use/configure VFs. not allowing the user
+to use some of the VFs for host networking.
+
+### Use Cases
+
+* As a user I want to use a virtual function for SDN network
+* As a user I want to create the virtual functions via nmstate
+* As a user I want pods to use virtual functions from a pre-configured PF 
+
+### Goals
+
+* Allow the SR-IOV operator to handle the configure and pod allocation of a subset of virtual functions.
+* Allow the user to Allocate the number of virtual functions he wants for the system and the subset he wants for pods
+* Not resetting the numOfVfs for PFs that the operator didn't configure
+
+### Non-Goals
+
+* Supporting switchdev mode (may change in the future if there is a request)
+
+## Proposal
+
+Create a flow in the SR-IOV operator where the user can request a configuration for a subset of virtual functions.
+
+The operator will first validate and the requested PF contains the requested amount of virtual functions allocated,
+Then the operator will configure the subset of virtual functions with the request driver and will update the device plugin
```

Comment:
what about MTU, GUID, MAC, driver unbind/bind for rdma?

can you add a list of the current operations sriov operator (and config daemon) performs on PF and VF 
and then list the set of operations it will perform on a VF that belongs to externally manager PF ?